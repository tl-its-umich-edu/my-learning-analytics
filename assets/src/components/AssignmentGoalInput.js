import React, { useEffect, useState, useRef } from 'react'
import { withStyles } from '@material-ui/core'
import Grid from '@material-ui/core/Grid'
import Button from '@material-ui/core/Button'
import StyledTextField from './StyledTextField'
import debounce from 'lodash.debounce'
import { eventLogExtra } from '../util/object'

const styles = ({
  goalGradeInput: {
    marginTop: 0,
    width: 150
  }
})

function AssignmentGoalInput (props) {
  const {
    currentGrade,
    maxPossibleGrade,
    goalGrade,
    setGoalGrade,
    setEventLog,
    eventLog,
    handleClearGoalGrades,
    mathWarning,
    classes
  } = props

  const [goalGradeInternal, setGoalGradeInternal] = useState(goalGrade)
  const debouncedGoalGrade = useRef(debounce(q => setGoalGrade(q), 500)).current
  const updateGoalGradeInternal = (grade) => {
    const v = { courseGoalGrade: grade }
    if (goalGrade !== '') {
      // only send prev grade object when there is previous value
      v.prevCourseGoalGrade = goalGrade
    }
    setEventLog(eventLogExtra(v, eventLog, currentGrade, maxPossibleGrade))
    debouncedGoalGrade(grade)
    setGoalGradeInternal(grade)
  }

  useEffect(() => {
    setGoalGradeInternal(goalGrade)
  }, [goalGrade])

  return (
    <Grid item>
      <StyledTextField
        error={goalGradeInternal > 100 || mathWarning || goalGradeInternal > maxPossibleGrade}
        id='standard-number'
        value={goalGradeInternal}
        label={
          mathWarning
            ? 'Math may no longer add up'
            : goalGrade > 100
              ? 'Over 100%'
              : goalGrade > maxPossibleGrade
                ? 'Greater than max possible grade'
                : 'Set Minimum Goal'
        }
        onChange={event => {
          const goalGrade = event.target.value
          if (goalGrade === '') {
            updateGoalGradeInternal('')
          } else if (goalGrade <= 0) {
            updateGoalGradeInternal(0)
          } else if (goalGrade > 125) {
            updateGoalGradeInternal(125)
          } else {
            updateGoalGradeInternal(goalGrade)
          }
        }}
        type='number'
        className={classes.goalGradeInput}
        margin='normal'
        variant='outlined'
        style={{ marginRight: '10px', width: '25ch' }}
      />
      {
        <Button
          variant='contained'
          className={classes.clearButton}
          onClick={handleClearGoalGrades}
          aria-label='clear'
        >
          Clear
        </Button>
      }
    </Grid>
  )
}

export default withStyles(styles)(AssignmentGoalInput)
