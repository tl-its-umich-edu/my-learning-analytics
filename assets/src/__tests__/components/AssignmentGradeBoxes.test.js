/* global it, expect */

import React from 'react'
import AssignmentGoalInput from '../../components/AssignmentGoalInput'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <AssignmentGoalInput
        currentGrade={85}
        goalGrade={90}
        maxPossibleGrade={95}
      />
    ).toJSON()

  expect(tree).toMatchSnapshot()
})
