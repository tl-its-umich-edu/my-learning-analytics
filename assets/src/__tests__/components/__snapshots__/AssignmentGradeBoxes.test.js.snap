// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container"
>
  <div
    className="MuiGrid-root MuiGrid-item"
    style={
      Object {
        "display": "inline-block",
      }
    }
  >
    <h6
      className="MuiTypography-root MuiTypography-h6"
      style={
        Object {
          "display": "inline-block",
          "marginRight": "10px",
        }
      }
    >
      Goal
    </h6>
    <div
      className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-2 AssignmentGoalInput-goalGradeInput-1 MuiFormControl-marginNormal"
      style={
        Object {
          "marginRight": "10px",
          "width": "25ch",
        }
      }
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
        data-shrink={true}
        htmlFor="standard-number"
        id="standard-number-label"
      >
        Set course goal
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input"
          disabled={false}
          id="standard-number"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="number"
          value={90}
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-5 PrivateNotchedOutline-legendNotched-6"
          >
            <span>
              Set course goal
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      aria-label="clear"
      className="MuiButtonBase-root MuiButton-root MuiButton-contained"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Clear
      </span>
    </button>
  </div>
</div>
`;
